File: Account.java
/**
 * Created by eamal27 on 2017-03-16.
 */
public class Account {
    public String accountType;
    public float creditAmount;
    public static String username;//causes problems to the test refund amount not updated, updateusersTest 
    public boolean deleted;

    public Account(){

    }
    public Account(String accountType, float creditAmount, String username) {
        this.accountType = accountType;
        this.creditAmount = creditAmount;
        this.username = username;
        this.deleted = false;
    }

    public float getCreditAmount(){
        return creditAmount;
    }

    public String getUsername(){
        return username;
    }

    public String getAccountType(){
        return accountType;
    }

    public boolean getDeleted(){
        return deleted;
    }
}

File: FrontEnd/frontend/src/Account.cpp
#include "Account.h"

using namespace std;

Account::Account(string username, float creditAmount, string accountType) {
    this->username = username;
    this->creditAmount = creditAmount;
    this->accountType = accountType;
    this->deleted = false;
    this->creditAdded = 0;
    this->creditReceived = 0;
}

void Account::addCredit(float creditAmount) {
    this->creditAmount += creditAmount;
}

void Account::subtractCredit(float creditAmount) {
    this->creditAmount -= creditAmount;
}

void Account::buy(string eventName, int ticketCount) {
    // TODO: implement so that it subtracts tickets and adds credit
}

float Account::getCreditAmount() {
    return this->creditAmount;
}

string Account::getUsername() {
    return this->username;
}

string Account::getAccountType() {
    return this->accountType;
}

bool Account::isAdmin() {
    return this->accountType == "AA";
}

void Account::deleteUser() {
    this->deleted = true;
}

bool Account::isDeleted() {
    return this->deleted;
}

float Account::getCreditAdded() {
    return this->creditAdded;
}

void Account::setCreditAdded(float amount) {
    this->creditAdded = amount;
}

float Account::getCreditReceived() {
    return this->creditReceived;
}

void Account::setCreditReceived(float amount) {
    this->creditReceived = amount;
}
